src/util/bip32.rs

```diff
@@ -589,7 +586,7 @@ impl ExtendedPrivKey {
                 network: network,
                 key: secp256k1::SecretKey::from_slice(
                     &data[46..78]
-                ).map_err(Error::Ecdsa)?,
+                ).map_err(Error::Secp256k1)?,
```

Comment:
Given that we have a `From<secp256k1::Error> for Error` implementation, I think this `map_err` is redundant because `?` calls `Error::from` for you.